src/strategies/sound-protocol-call-tokenuri/transformer.mjs

```diff
@@ -0,0 +1,92 @@
+// @format
+import { decodeParameters } from "eth-fun";
+import logger from "../../logger.mjs";
+
+export const name = "sound-protocol-call-tokenuri";
+export const version = "0.1.0";
+
+const resultKey = "tokenURI";
+const log = logger(name);
+
+const extractTokenId = (tokenURI) => parseInt(tokenURI.split("/")[3]);
+
+export function onClose() {
+  return;
+}
+
+export function onError(error) {
+  log(error.toString());
+  throw error;
+}
+
+export function onLine(line) {
+  let obj;
+
+  try {
+    obj = JSON.parse(line);
+  } catch (err) {
+    log(err.toString());
+    return;
+  }
+
+  let { results: tokenURIs } = obj;
+  let result = [];
+  try {
+    tokenURIs = tokenURIs.filter(
+      (tokenURI) => tokenURI.split("/").length === 4
+    );
+  } catch (err) {
+    log(err.toString());
+    return;
+  }
+
+  try {
+    tokenURIs.sort((a, b) => extractTokenId(a) - extractTokenId(b));
```

Comment:
@TimDaub Do you mean this part? I sorted array by tokenId to get exact range of tokenIds for each song. The result what extractor returned is not sorted (because of worker's parallelism processing). So it is too difficult to confirm range of tokenIds. Of course, if the sound edition contract include only one song, it is no problem. I only should find minimum and maximum tokenId out. But to prevent exceptions such as Reo and more in the future, I am trying to find exact range of tokenId and for this, I sorted request array.

So the result of transformer is something like that:

```
[
  {
    url: "-bdk22323....",
    min: [1, 40],
    max: [30, 321]               // 1 - 30 and 40 - 321
  },
  {
    url: "d2dadk22323....",
    min: [31],
    max: [39]                       // 31 - 39
  },
]
```

```
tokenURI: [
  ar://-bdk22323..../[1-30, 40-321],
  ar://d2dadk22323../[31-39]
]
```

Please correct me if my thought is wrong